R/toast.R

```diff
@@ -0,0 +1,466 @@
+#' Toast notifications
+#'
+#' @description
+#' Toast notifications are lightweight, temporary messages designed to mimic
+#' push notifications from mobile and desktop operating systems. They are built
+#' on [Bootstrap 5's toast
+#' component](https://getbootstrap.com/docs/5.3/components/toasts/).
+#'
+#' bslib includes a complete example of toasts and their many configuration
+#' options:
+#'
+#' ```r
+#' shiny::runExample("toast", package = "bslib")
+#' ```
+#'
+#' @examplesIf rlang::is_interactive()
+#' library(shiny)
+#' library(bslib)
+#'
+#' ui <- page_fluid(
+#'   actionButton("show_simple", "Simple Toast"),
+#'   actionButton("show_header", "Toast with Header")
+#' )
+#'
+#' server <- function(input, output, session) {
+#'   observeEvent(input$show_simple, {
+#'     show_toast(
+#'       toast(
+#'         body = "Operation completed successfully!",
+#'         header = "Success",
+#'         type = "success"
+#'       )
+#'     )
+#'   })
+#'
+#'   observeEvent(input$show_header, {
+#'     show_toast(
+#'       toast(
+#'         body = "Your settings have been saved.",
+#'         header = toast_header(
+#'           title = "Settings Updated",
+#'           status = "just now"
+#'         ),
+#'         type = "success"
+#'       )
+#'     )
+#'   })
+#' }
+#'
+#' shinyApp(ui, server)
+#'
+#' @param ... Body content of the toast. Can be a string, or any HTML elements.
+#'   Named arguments will be treated as HTML attributes for the toast container.
+#' @param header Optional header content. Can be a string, or the result of
+#'   [toast_header()]. If provided, creates a `.toast-header` with close button
+#'   (if `closable = TRUE`).
+#' @param id Optional unique identifier for the toast. If `NULL`, an ID will be
+#'   automatically generated when the toast is shown via [show_toast()].
+#'   Providing a stable ID allows you to hide the toast later. If a toast with
+#'   `id` is already visible, that toast is automatically hidden before showing
+#'   the new toast with the same `id` so that only one toast with a given ID is
+#'   shown at once.
+#' @param type Optional semantic type. One of `NULL`, `"primary"`,
+#'   `"secondary"`, `"success"`, `"info"`, `"warning"`, `"danger"`, `"light"`,
+#'   or `"dark"`. Applies appropriate Bootstrap background utility classes
+#'   (`text-bg-*`).
+#' @param autohide_s Numeric. Number of seconds after which the toast should
```

Comment:
I find myself wanting to call this `duration`, partly because of the `showNotification(duration = )` precedence?